Frameworks/Foundation/NSException.mm

```diff
@@ -216,7 +244,18 @@ - (HRESULT)_hresult {
         return [hresultValue unsignedIntValue];
     }
 
-    TraceWarning(L"NSException", L"Called -hresult on an NSException without an HRESULT!");
-    return E_UNEXPECTED;
+    return [[self class] _HRESULTForExceptionName:self.name];
+}
+
+- (void)_processException {
+    HRESULT hr = [self _hresult];
+    HSTRING_HEADER header;
+    HSTRING hsReason = nullptr;
+    
+    if(self.reason != nil) {
+        LPCWSTR reason = (LPCWSTR)([[self reason] cStringUsingEncoding:NSUnicodeStringEncoding]);
+        WindowsCreateStringReference(reason, wcslen(reason), &header, &hsReason);
```

Comment:
The "jist" of it is that you should be using **Wrappers::HStringReference** instead of manually creating one and having to worry about its lifetime. 

I totally understand that "reason" is the backing store but that then begs the question of why don't need to delete the HSTRING that is presumably pointing to it. The interesting bit is the HEADER that is on the stack and that under the covers that header is being cast to the output HSTRING hsReason. This means that the storage for the HSTRING itself is just the header on the stack. 

The fact though that you had to read the remarks, and still weren't 100% certain what was going on here suggests that a lifetime managed object like the wrappers one will make the code more maintainable, less verbose, and all around more clear what you are intending. 

---
In reply to: [93299684](https://github.com/Microsoft/WinObjC/pull/1570#discussion_r93299684) [](ancestors = 93299684,93297284,93125423)